### PR TITLE
improved framebuffer for deffered rendering

### PR DESCRIPTION
**Pull request by nem0**

### CODE REVIEW

File: src/animation/animation.cpp [hunk 1]
```diff
@@ -56,21 +58,18 @@ Animation::~Animation()
...
-static const float ANIMATION_FPS = 30.0f;
-
-
 void Animation::getPose(float time, Pose& pose, Model& model) const
 {
 	PROFILE_FUNCTION();
 	if(model.isReady())
 	{
-		int frame = (int)(time * ANIMATION_FPS);
+		int frame = (int)(time * m_fps);
```
Comment (tluqo) [inline src/animation/animation.cpp]: does frame have to be an integer? you do multiple time conversion int <-> float. at least in this function you could do only one:  float frame_count = float(m_frame_count);

File: src/animation/animation.h [hunk 2]
```diff
@@ -59,6 +59,7 @@ class LUMIX_ANIMATION_API Animation : public Resource
 		Vec3* m_positions;
 		Quat* m_rotations;
 		uint32_t* m_bones;
+		int m_fps;
```
Comment (tluqo) [inline src/animation/animation.h]: does it have to be an integer?

File: src/editor/world_editor.cpp [hunk 3]
```diff
@@ -289,9 +448,10 @@ class RemoveArrayPropertyItemCommand : public IEditorCommand
 		}
 
 	private:
+		WorldEditor& m_editor;
 		Component m_component;
 		int m_index;
-		IArrayDescriptor& m_descriptor;
+		const IArrayDescriptor* m_descriptor;
```
Comment (tluqo) [inline src/editor/world_editor.cpp]: since m_descriptor is pointer now, you should assert not null on the every place you use it.

**Comment by tluqo** [comment]
Ship it!

File: src/animation/animation.cpp (hunk 1 above)
Comment (nem0) [inline src/animation/animation.cpp]: It is used as an index in array

Comment (tluqo) [inline src/animation/animation.cpp]: I see, off2 :-/

File: src/animation/animation.h (hunk 2 above)
Comment (nem0) [inline src/animation/animation.h]: it's used once as an integer and once as a float, so it does not matter from performance view

**Pull request merged**